webrtc/RTCPeerConnection-ondatachannel.html

```diff
@@ -5,180 +5,388 @@
+promise_test(async (t) => {
+  const resolver = new Resolver();
+  const pc1 = new RTCPeerConnection();
+  const pc2 = new RTCPeerConnection();
+  t.add_cleanup(() => pc1.close());
+  t.add_cleanup(() => pc2.close());
 
-    localPc.createDataChannel('test');
-
-    remotePc.addEventListener('datachannel', onDataChannel);
-    exchangeIceCandidates(localPc, remotePc);
-    doSignalingHandshake(localPc, remotePc);
-  }, 'datachannel event should fire when new data channel is announced to the remote peer');
-
-  /*
-    6.2.  RTCDataChannel
-      interface RTCDataChannel : EventTarget {
-        readonly attribute USVString           label;
-        readonly attribute boolean             ordered;
-        readonly attribute unsigned short?     maxPacketLifeTime;
-        readonly attribute unsigned short?     maxRetransmits;
-        readonly attribute USVString           protocol;
-        readonly attribute boolean             negotiated;
-        readonly attribute unsigned short?     id;
-        readonly attribute RTCPriorityType     priority;
-        readonly attribute RTCDataChannelState readyState;
-        ...
-      };
-
-      When an underlying data transport is to be announced
-        3.  Let configuration be an information bundle received from the
-            other peer as a part of the process to establish the underlying
-            data transport described by the WebRTC DataChannel Protocol
-            specification [RTCWEB-DATA-PROTOCOL].
-        4.  Initialize channel's label, ordered, maxPacketLifeTime,
-            maxRetransmits, protocol, negotiated and id attributes to their
-            corresponding values in configuration.
-   */
-  async_test(t => {
-    const localPc = new RTCPeerConnection();
-    t.add_cleanup(() => localPc.close());
-    const remotePc = new RTCPeerConnection();
-
-    t.add_cleanup(() => remotePc.close());
-
-    const onDataChannel = t.step_func_done(event => {
-      const remoteChannel = event.channel;
-      assert_true(remoteChannel instanceof RTCDataChannel,
-        'Expect channel to be instance of RTCDataChannel');
-
-      assert_equals(remoteChannel.label, 'test');
-      assert_equals(remoteChannel.id, 8);
-      assert_equals(remoteChannel.ordered, false);
-      assert_equals(remoteChannel.maxRetransmits, 1);
-      assert_equals(remoteChannel.protocol, 'custom');
-      assert_equals(remoteChannel.priority, 'high');
+  const message = 'meow meow!';
+
+  pc2.ondatachannel = t.step_func((event) => {
+    const dc2 = event.channel;
+    dc2.send(message);
+  });
+
+  const dc1 = pc1.createDataChannel('fire-me!');
+  dc1.onmessage = t.step_func((event) => {
+    assert_equals(event.data, message,
+      'Received data should be equal to sent data');
+
+    resolver.resolve();
+  });
+
+  exchangeIceCandidates(pc1, pc2);
+  await doSignalingHandshake(pc1, pc2);
+  t.step_timeout(() => {
+    assert_unreached('Data should have been received by now (timeout reached)');
+  }, 500);
```

Comment:
Ditto here and below.

Reply:
Will remove these.
If we want to add back something, we should probably make doSignalingHandshake reject after a timeout.